cargo-quickinstall/src/utils.rs

```diff
@@ -0,0 +1,14 @@
+use home::cargo_home;
+use std::{io, path::PathBuf};
+
+pub fn utf8_to_string_lossy(bytes: Vec<u8>) -> String {
+    String::from_utf8(bytes)
+        .unwrap_or_else(|err| String::from_utf8_lossy(err.as_bytes()).into_owned())
+}
+
+pub fn get_cargo_bin_dir() -> io::Result<PathBuf> {
+    cargo_home().map(|mut cargo_bin_dir| {
+        cargo_bin_dir.push("bin");
+        cargo_bin_dir
+    })
```

Comment:
(optional) I tend to find ? and ok-wrapping easier to read. The difference is more pronounced when things get more complicated (like with the get_owned() and_then() chain above), but consistency is also valuable.
```suggestion
    let mut cargo_bin_dir = cargo_home()?;
    cargo_bin_dir.push("bin");
    Ok(cargo_bin_dir)
```

also we don't care about performance here, so join is probably fine:

```suggestion
    let cargo_bin_dir = cargo_home()?;
    Ok(cargo_bin_dir.join("bin"))
```